builder/azure/arm/config.go

```diff
@@ -303,6 +303,8 @@ func setCloudEnvironment(c *Config) error {
 	switch name {
 	case "CHINA", "CHINACLOUD", "AZURECHINACLOUD":
 		c.cloudEnvironment = &azure.ChinaCloud
+	case "GERMAN", "GERMANCLOUD", "AZUREGERMANCLOUD":
```

Comment:
I assume these all mirror what the xplat CLI names them?
Should we push for code like this in Azure SDK?